src/main/java/org/wildfly/security/auth/realm/ldap/X509EvidenceVerifier.java

```diff
@@ -124,7 +125,9 @@ public boolean verifyCertificate(X509Certificate certificate, Attributes attribu
             Attribute attribute = attributes.get(ldapAttribute);
             final int size = attribute.size();
             for (int i = 0; i < size; i++) {
-                if (certificate.getSubjectDN().getName().equals(attribute.get(i))) {
+                LdapName certSubjectDn = new LdapName(certificate.getSubjectDN().getName());
```

Comment:
First, ``X509Certificate#getSubjectDN()`` is "denigrated", so we should probably not be using it.

Apart from that, I'm unsure if it is better to use an LdapName here or a X500Principal like this:
```
X500Principal certSubjectDn = certificate.getSubjectX500Principal();
X500Principal attributeDn = new X500Principal((String) attribute.get(i));
```
The difference is probably not major, but if we're already using ``getSubjectX500Principal()`` then we might as well perform all comparisons in terms of ``X500Principal``.